TM1py/Services/CellService.py

```diff
@@ -991,6 +1012,163 @@ def write_through_unbound_process(self, cube_name: str, cellset_as_dict: Dict, i
         if not all(successes):
             raise TM1pyWritePartialFailureException(statuses, log_files, len(successes))
 
+    @require_admin
+    @manage_transaction_log
+    @require_pandas
+    def write_through_blob(self, cube_name: str, cellset_as_dict: dict, increment: bool = False,
+                           sandbox_name: str = None, skip_non_updateable: bool = False,
+                           folder_name: str = "tm1py", delete_blob=True, dimensions: str = None,
+                           **kwargs):
+        """
+        Writes data back to TM1 via an unbound TI process having an uploaded CSV as data source
+        :param cube_name: str
+        :param cellset_as_dict:
+        :param increment: increment or update cell values
+        :param sandbox_name: str
+        :param skip_non_updateable skip cells that are not updateable (e.g. rule derived or consolidated)
+        :param folder_name
+        :param delete_blob: choose False for debugging purposes
+        :param dimensions: optional. Dimension names in their natural order. Will speed up the execution!
+        :param kwargs:
+        :return: Success: bool, Messages: list, ChangeSet: None
+        """
+
+        application_service = ApplicationService(self._rest)
+        process_service = ProcessService(self._rest)
+        cube_service = self.get_cube_service()
+
+        # Generate hash based on tm1-session-id and local-thread guarantee unique name
+        unique_string = f"{self._rest.session_id}{threading.get_ident()}"
+        unique_hash = hashlib.sha256(unique_string.encode('utf-8')).hexdigest()[:12]
+
+        # Transform the cellset into a dictionary and export to CSV
+        csv_content = StringIO()
+        csv_writer = csv.writer(
+            csv_content,
+            delimiter=",",
+            quoting=csv.QUOTE_ALL)
+        csv_writer.writerows(
+            list(elements) + [value.replace('\r', '').replace('\n', '') if isinstance(value, str) else value]
+            for elements, value
+            in cellset_as_dict.items())
+
+        # Define the path and file name
+        filename = f'{unique_hash}.csv'
+
+        # Upload CSV to TM1 server using ApplicationService:
+        # Create a folder TM1py
+        tm1py_folder = FolderApplication(path='', name=folder_name)
```

Comment:
I'm wondering if uploading to the 'Blobs' folder instead of 'Applications' would be a cleaner approach here? 

The main reason being is that in the `_create_blob_write_process` function, the }Externals folder is being directly referenced. It feels like applications residing in the }Externals folder is an implementation detail of TM1 that TM1py shouldn't care about. If the document was uploaded through the 'Blobs' folder, the id of the blob can be passed directly to the datasource parameters e.g.

`datasource_data_source_name_for_server=blob_filename,
datasource_data_source_name_for_client=blob_filename`

Reply:
I think you have a point there @gbryant-dev.

We should change it to use blobs and ignore the concepts of TM1 applications.

Reply:
@MariusWirtz Did you get a chance to test this in a Linux env?

Reply:
good catch @rkvinoth. It did not work in a Linux env. 

After we changed to using blobs it does.